subsys/fb/cfb.c

```diff
@@ -103,6 +103,46 @@ static u8_t draw_char_vtmono(const struct char_framebuffer *fb,
 	return fptr->width;
 }
 
+/*
+ * Draw the RGB character in the rgb565 linear framebuffer
+ */
+static u8_t draw_char_vtrgb565(const struct char_framebuffer *fb,
+			     char c, u16_t x, u16_t y)
+{
+	const struct cfb_font *fptr = &(fb->fonts[fb->font_idx]);
+	u8_t *glyph_ptr;
+
+	if (c < fptr->first_char || c > fptr->last_char) {
+		c = ' ';
+	}
+
+	glyph_ptr = get_glyph_ptr(fptr, c);
+	if (!glyph_ptr) {
+		LOG_DBG("can not find the char %c", c);
+		return 0;
+	}
+
+	for (size_t g_x = 0; g_x < fptr->width; g_x++) {
+		u8_t masker = 1 << (fptr->width - g_x - 1);
+
+		for (size_t g_y = 0; g_y < fptr->height; g_y++) {
+			u8_t v = 0;
+			u16_t *ppixel = NULL;
+			u32_t fb_y = (y + g_y) * fb->x_res;
+
+			if (2 * (fb_y + x + g_x) >= fb->size) {
+				return 0;
+			}
+			ppixel = (u16_t *)&fb->buf[2 * (fb_y + x + g_x)];
+			v = (glyph_ptr[g_y] & masker) == 0 ? 0 : 255;
+			*ppixel =  ((v>>3)<<11) | ((v>>2)<<5) | (v>>3);
```

Comment:
This seems a little complex.  Isn't it equivalent to:
```
*ppixel = ((glyph_ptr[g_y] & masker) == 0) ? 0 : 0xFFFF;
```
?

Reply:
no, this is to convert the 8x8 mono to rgb565 format, here I use v to be r(5bit )g(6 bit)b(5bit).

Reply:
The right hand side converts to either 0 or 0xFFFF depending on whether the masker bit is set.  It's too complex.  Add a comment indicating that 0xFFFF means white in RGB565 encoding, if you want, but don't make maintainers try to figure out what that expression means.

Reply:
good suggestion, add one comment